test/DePayRouterV1CurveFiSwap01.spec.ts

```diff
@@ -0,0 +1,63 @@
+import chai, { expect } from 'chai'
+
+import { solidity, loadFixture } from 'ethereum-waffle'
+import { BigNumber } from 'ethers'
+
+import {
+  cureFiFixture,
+  cureFiSwapFixture,
+  curveFiSystemFixture,
+  unapprovedCureFiFixture
+} from './fixtures/curvefi-swap'
+
+import { route as routerFunc } from './functions'
+
+import { now, ETH } from './utils'
+
+chai.use(solidity)
+
+describe('CurveFiSwap01', () => {
+  it('AddressProvider and Registry should be deploy correctly', async () => {
+    await loadFixture(unapprovedCureFiFixture)
+  })
+
+  it('Admin address of AddressProvider should be ownerWallet', async () => {
+    const { curveFiAddressProvider, ownerWallet } = await loadFixture(unapprovedCureFiFixture)
+    expect(ownerWallet.address).to.equal(await curveFiAddressProvider.admin())
+  })
+
+  it("Registry's address in AddressProvider should be the same", async () => {
+    const { curveFiAddressProvider, curveFiRegistryMock } = await loadFixture(unapprovedCureFiFixture)
+    expect(curveFiRegistryMock.address).to.equal(await curveFiAddressProvider.get_registry())
+  })
+
+  it('Index of tokens is should be correct in the registry', async () => {
+    const { curveFiRegistryMock, fromToken, toToken, curveFiPoolMock } = await loadFixture(cureFiSwapFixture)
+    const [fromIndex, toIndex, isUnderlying] = await curveFiRegistryMock.get_coin_indices(
+      curveFiPoolMock.address,
+      fromToken.address,
+      toToken.address
+    )
+    expect(fromIndex.toNumber()).to.equal(0)
+    expect(toIndex.toNumber()).to.equal(1)
+    expect(isUnderlying).to.equal(false)
+  })
+
+  it('Swap token via CurveFiSwap01 plugin', async () => {
```

Comment:
We need to add at least one test case where we swap + pay. See Uniswap example here: https://github.com/DePayFi/depay-ethereum-router/blob/master/test/DePayRouterV1Uniswap01.spec.ts#L52